File: FE/AL-homepage/src/types/curriculumDetail.ts
import type { Todo } from './todo'
import type { Homework } from './homework'

export interface CurriculumDetail {
  curriculum_id: number
  subject_id: number
  day: number
  progress: string
  curri_detail_id: number
  notice?: string
  homework?: string
  before_class?: string
  in_class?: string
  todos?: Todo[]
  homeworks?: Homework[]
}

export interface CreateCurriculumDetailRequest {
  curriculum_id: number
  subject_id: number
  day: number
  progress: string
}

export interface UpdateCurriculumDetailRequest {
  curriculum_id?: number
  subject_id?: number
  day?: number
  progress?: string
}

export interface GetAllCurriculumDetailsParams {
  skip?: number
  limit?: number
  curriculum_id?: number
  subject_id?: number
}

File: BE/avalon/app/models/curriculum_detail.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from .base import Base

class CurriculumDetail(Base):
    __tablename__ = "curriculum_detail"
    curri_detail_id = Column(Integer, primary_key=True, autoincrement=True)
    curriculum_id = Column(Integer, ForeignKey("curriculums.curriculum_id"), nullable=False)
    subject_id = Column(Integer, ForeignKey("subjects.subject_id"), nullable=False)
    day = Column('DAY', Integer, nullable=False)
    progress = Column(String(100), nullable=False)

    # Relationships
    homeworks = relationship("Homework", back_populates="curriculum_detail")
